hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/impl/connector/kafka/StreamKafkaP.java

```diff
@@ -91,53 +96,64 @@
     protected void init(@Nonnull Context context) throws Exception {
         snapshottingEnabled = context.snapshottingEnabled();
         consumer = new KafkaConsumer<>(properties);
-        reassignPartitions();
+        reassignPartitions(true);
```

Comment:
wouldn't it make more sense to do partition assignment in the Supplier/MetaSupplier. It looks like we're relying on each node independently deciding what partitions it should subscribe to?

Reply:
1. you only know the partitions when you have a `consumer`. You don't have one in Supplier. 
2. we regularly check for new partitions, kafka supports adding partitions during runtime (unlike HZ)